src/internal/channel.js

```diff
@@ -113,7 +113,7 @@ export function channel(buffer) {
 }
 
 export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
-  if(arguments.length > 1)
+  if(typeof matcher !== 'undefined')
```

Comment:
That will fix the issue I was having, which was passing 2 arguments without a `matcher`. However, I believe it is idiomatic to test arguments for `undefined` instead of testing number of arguments.

With the introduction of default arguments, there is now a reason to intentionally pass `undefined` arguments in order to use default values and still pass an argument after the defaulted one. This particular function is a great example: it has a default argument in the middle position, which means I have to intentionally pass in `undefined` as `buffer` if I wanted to take advantage of the default and still pass in a `matcher`.

BTW the `matcher` argument is [not documented](https://github.com/yelouafi/redux-saga/tree/master/docs/api#eventchannelsubscribebuffer) so we should probably fix the doc as well, but I don't know enough about it to fix the doc for you.